cpp/src/rolling/rolling.cu

```diff
@@ -209,35 +211,35 @@ struct rolling_window_launcher
                                      WindowIterator preceding_window_begin,
                                      WindowIterator following_window_begin,
                                      size_type min_periods,
-                                     rolling_operator op,
+                                     std::unique_ptr<aggregation> const& aggr,
                                      rmm::mr::device_memory_resource *mr,
                                      cudaStream_t stream)
   {
-    switch (op) {
-    case rolling_operator::SUM:
-      return dispatch_aggregation_type<T, cudf::DeviceSum,
-                                       rolling_operator::SUM>(input, preceding_window_begin,
+    switch (aggr->kind) {
```

Comment:
You can use the `aggregation_dispatcher` for this instead https://github.com/rapidsai/cudf/blob/branch-0.12/cpp/include/cudf/detail/aggregation/aggregation.hpp#L186